frontend/individual_pedigree.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file This file contains the Pedigree function. This function fetches
+ *       pedigree for a given `id` (parsed from the url).
+ */
+
+import React, { Component, useEffect } from 'react'
+import { Link } from "react-router-dom";
+import { get } from './communication';
+import { PedigreeNetwork } from "./pedigree"
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableRow from '@material-ui/core/TableRow';
+
+
+/**
+ * Shows the information of a given individual and the pedigree graph built using the vis-network component
+ */
+export function IndividualPedigree({ id, generations }: { id: string }) {
+  const [pedigree, setPedigree] = React.useState(undefined as any)
+  const [individual, setIndividual] = React.useState(undefined as any)
+  const [generations_input, setGenerations] = React.useState(generations)
```

Comment:
Note that generations_input won't change if the generations prop (input) is changed (from parent).

One way to sync them up is to add an effect depending on `generations` that sets this state to `generations` when it changes.

Reply:
But the generations prop is not expected to change, this value is read from the url parameters, like the id in the previous comment. I thought this dependency was meant to be used for a changing parameter.

Reply:
ok

Reply:
Right one can have props that are not supposed to change. One way to enforce that they don't change is to remount the component using `key`: 
https://www.saltycrane.com/blog/2019/12/how-remount-react-component-when-prop-changes/
I think this is slightly weird since this is determined at the call site of the component rather than at the definition site of the component. It would make sense if one could indicate this when defining the component.